This PR is for:

carlosveigadev opened the pull request:
- Add Friendship model;
- Create friendship methods;
- Add RSpec tests.

sanadwj committed:
Create a friendships model

Add association between user and friends

Add rspec and capybara gem for testing

carlosveigadev committed:
add RSpec and Capybara config

add association rspec tests to friendship model

carlosveigadev committed:
fix rubocop offenses

carlosveigadev committed:
test github actions

carlosveigadev committed:
add rubocop linters

carlosveigadev committed:
fix rubocop file to work on githubactions

carlosveigadev committed:
fix rubocop offenses

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello Carlos Da Veiga & Sanad Abujbara** 
Great job with your first submission :muscle:  :muscle: 
We still need to fix some issues:

- [x] Posts timeline should only include posts written by logged-in user and their friends
- [x] User should be able to request friendship to any other user
- [x] User must be capable see friendships requests
- [x] User must be capable see friendships requests
- [x] User has to accept a friendship request to start a friendship 
- [x] User must be capable to reject a friendship request
- [x] There must be a button to invite the user to friendship on users list page and on single-user page
- [x] Fix rubocop offenses
- [ ] add tests for the models and views

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

sanadwj committed:
Add frienships controller

carlosveigadev committed:
add friendship create method

update user.html layout

sanadwj committed:
Add a friend request logic

carlosveigadev committed:
add pseudocode to friendship controller

sanadwj committed:
Add accept frien request for the user

carlosveigadev committed:
add request layouts to user show html

carlosveigadev committed:
connect user show with friendship controller

sanadwj committed:
Add buttons to Add friend and See profile

Import bootstrap

carlosveigadev committed:
add reject logic to friendship controller

update users layout

carlosveigadev committed:
start logic of post of only friends

sanadwj committed:
Add helper methods for user

Finish the logic to just sho the user anf friends the timeline

Add cancel button for pending friends

carlosveigadev committed:
fix add friend button to work properly

add some RSPec tests

sanadwj committed:
Add friendship views tests

Add comment & friendship tests

Add unfriend button

carlosveigadev committed:
add post feature rspec

fix rubocop offenses

sanadwj commented:
Thank you @andresporras3423 for the review.
We fixed all the improvements that you suggested.

Dandush03 reviewed (approved):
# Status: :rocket: Approved :rocket: 
### Great Work Implementing Changes :muscle: 
# Happy Coding :tophat:
